misc/pytorch_toolkit/chest_xray_screening_federated_gcn/README.md

```diff
@@ -0,0 +1,142 @@
+# Federated Learning for Site Aware Chest Radiograph Screening
+
+The shortage of Radiologists is inspiring the development of
+Deep Learning (DL) based solutions for detecting cardio, thoracic and pulmonary pathologies in Chest radiographs through
+multi-institutional collaborations. However, sharing the training data across multiple sites is often impossible due to privacy, ownership and technical challenges. Although Federated
+Learning (FL) has emerged as a solution to this, the large
+variations in disease prevalence and co-morbidity distributions
+across the sites may hinder proper training. We propose a DL
+architecture with a Convolutional Neural Network (CNN) followed by a Graph Neural Network (GNN) to address this issue.
+The CNN-GNN model is trained by modifying the Federated
+Averaging algorithm. The CNN weights are shared across all
+sites to extract robust features while separate GNN models are
+trained at each site to leverage the local co-morbidity dependencies for multi-label disease classification. The CheXpert
+dataset is partitioned across five sites to simulate the FL set
+up. Federated training did not show any significant drop in
+performance over centralized training. The site-specific GNN
+models also demonstrated their efficacy in modelling local disease co-occurrence statistics leading to an average area under
+the ROC curve of 0.79 with a 1.74% improvement.
+
+Figure below shows the overall schematic diagram of federated learning proposed in <a href="#comp_journal">[1]</a>
+<img src = "./media/scheme.png" width=650>
+
+## The proposed CNN-GNN architecture 
+
+<img src = "./media/architecture.jpeg" width=650>
+
+Separate Fully Connected (FC) layers are employed to obtain different 512-D
+features for each class. These are used as node features to construct a graph whose edges capture the co-occurrence dependencies
+between the classes at each site. The graph is provided as input to a Graph Neural Network to obtain the prediction labels for
+each node. The entire CNN-GNN architecture is trainable in an end-to-end manner.
+
+## Results
+
+The overall performance of the proposed CNN-GNN is 
+<img src = "./media/results.jpeg" width=650>
+
+## Model
+
+Download `.pth` checkpoint for CNN-GNN model trained on CheXpert dataset with the following [link](http://kliv.iitkgp.ac.in/projects/miriad/model_weights/bmi34/high_low/weights.zip).
+
+Inference models will be made available in the [open_model_zoo](https://github.com/openvinotoolkit/open_model_zoo/tree/master/models/public) as well.
+
+Note: The ONNX and IR representation models accepts inputs of fixed size mentioned in configuration file. This needs to be updated based on the input size.
+
+## Setup
+
+* Ubuntu 20.04
+* Python 3.8
+* NVidia GPU for training
+* 16 GB RAM for inference
+
+## Code and Directory Organisation
+
+
+```
+radiology_compression/
+	src/
+       utils/
+        dataprep.py
+        downloader.py
```

Comment:
Update the entire code directory written in the Readme with the corresponding files of this code.